### PR TITLE
Jeremy/eng 998 general info

### PR DESCRIPTION
**Pull request by jerminatorhits**
## Description
Why did you write this code?
code ⛳️ 

## How to Test
- [ ] Visit localhost:4200/work/account
- [ ] Edit user info successfully via the form

Which devices did you test on?
- [x] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
Will this have future work?
Tried to implement a 'reset' button as well but our current `Textarea` using draft js doesn't update it's EditorState. May consider a refactor PR or simply use another wysiwyg (mobiledoc?)

## Learnings
Did you learn anything here you want to share with the team?
Combining libraries (e.g. Formik + Reactstrap) could be constraining, however, one thing that was pretty cool was Formik's `<Field />` component can be injected in Reactstrap's `<Input />` via a `tag` props like so:
```
<Input
  id={UserField.LAST_NAME}
  invalid={!!errors.lastName && !!touched.lastName}
  name={UserField.LAST_NAME}
  placeholder="Last name"
  tag={Field}
  type="text" />
```
it'll inherit the properties that come with [Field](https://jaredpalmer.com/formik/docs/api/field).

### CODE REVIEW

File: src/pages/Account/Account.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        
+        const { creditBalance, user } = data;
+        return (
+          <Container>
+            <h1>Profile</h1>
+            <hr />
+            <Nav tabs>
+              <NavItem>
+                <NavLink
+                  activeClassName="active"
```
Comment (kcvan) [inline src/pages/Account/Account.tsx]: should we move these details to an array of objects?

File: src/pages/Account/AccountGeneral.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  function handleSubmit(values: any, actions: FormikActions<any>) {
+    updateUser(values)
+      .then(() => {
+        alert('user successfully updated');
+      })
+      .catch((error: any) => {
+        console.log('error: ', error);
+        alert(error);
+      })
+      .finally(() => actions.setSubmitting(false));
```
Comment (kcvan) [inline src/pages/Account/AccountGeneral.tsx]: wooot `finally`

File: src/pages/Account/AccountGeneral.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+          </Row>
+        </Form>
+      )}
+    </Formik>
+  );
+
+  function handleSubmit(values: any, actions: FormikActions<any>) {
+    updateUser(values)
+      .then(() => {
+        alert('user successfully updated');
```
Comment (kcvan) [inline src/pages/Account/AccountGeneral.tsx]: i think we can use reactstrap's `<Alert />` component for all of our alerts now

File: src/pages/Account/AccountVerification.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+
+class AccountVerification extends React.Component<any> {
+  render() {
+    return (
+      <h1>This is the Verification Page</h1>
```
Comment (kcvan) [inline src/pages/Account/AccountVerification.tsx]: is it now

File: src/pages/Account/AccountGeneral.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+              }}
+              placeholder="Tell us about yourself"
+              value={values.about} />
+            <FormFeedback>{errors.about}</FormFeedback>
+          </FormGroup>
+
+          <hr />
+
+          <Row className="align-items-center justify-content-end">
+            <Col xs="auto" className="text-right float-right">
```
Comment (kcvan) [inline src/pages/Account/AccountGeneral.tsx]: we can remove `float` and `auto` here

File: src/pages/Account/Account.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  );
+}
+
+export default () => (
+  <FirebaseConsumer>
+    {({ user, loading }: FirebaseUserProps) => {
+      if (loading) {
+        return null;
+      }
+      return user ? <Account /> : <Redirect to="/login" />;
```
Comment (kcvan) [inline src/pages/Account/Account.tsx]: `/work/login`

Comment (kcvan) [reply src/pages/Account/Account.tsx]: for now

File: src/pages/Account/Account.tsx (hunk 1 above)
Comment (jerminatorhits) [reply src/pages/Account/Account.tsx]: done

File: src/pages/Account/Account.tsx (hunk 6 above)
Comment (jerminatorhits) [reply src/pages/Account/Account.tsx]: good catch!

File: src/pages/Account/AccountVerification.tsx (hunk 4 above)
Comment (jerminatorhits) [reply src/pages/Account/AccountVerification.tsx]: si senõr

File: src/pages/Account/AccountGeneral.tsx (hunk 5 above)
Comment (jerminatorhits) [reply src/pages/Account/AccountGeneral.tsx]: removed

File: src/pages/Account/AccountGeneral.tsx (hunk 3 above)
Comment (jerminatorhits) [reply src/pages/Account/AccountGeneral.tsx]: implemented. It looks pretty craze-mazing

**Pull request merged**